code/modules/vending/_vending.dm

```diff
@@ -552,22 +571,29 @@ GLOBAL_LIST_EMPTY(vending_products)
 				vend_ready = 1
 				return
 			var/datum/bank_account/account = C.registered_account
-			if(!account.adjust_money(-chef_price))
-				say("You do not possess the funds to purchase this meal.")
-		var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_SRV)
-		if(D)
-			D.adjust_money(chef_price)
-		use_power(5)
-
-		vending_machine_input[N] = max(vending_machine_input[N] - 1, 0)
-		for(var/obj/O in contents)
-			if(O.name == N)
-				say("Thank you for buying local and purchasing [O]!")
-				O.forceMove(drop_location())
-				break
-		vend_ready = 1
-		updateUsrDialog()
-		return
+			for(var/obj/O in contents)
+				if(O.name == N)
+					price = O.custom_price
+					break
+			if(!account.has_money(price))
+				say("You do not possess the funds to purchase this.")
+			else
+				account.adjust_money(-price)
+				var/datum/bank_account/D = private_a
+				if(D)
+					D.adjust_money(price)
+				use_power(5)
+
+				vending_machine_input[N] = max(vending_machine_input[N] - 1, 0)
+				for(var/obj/O in contents)
```

Comment:
Don't do this. You already did this exact loop a few lines ago. Instead just cache the item and check if it's still in the vendor here.